scripts/cmake/vcpkg_fail_port_install.cmake

```
@@ -40,23 +40,42 @@ Library linkage for which the build should fail early.
 #]===]
 
 function(vcpkg_fail_port_install)
-    cmake_parse_arguments(PARSE_ARGV 0 "arg" "ALWAYS" "MESSAGE" "ON_TARGET;ON_ARCH;ON_CRT_LINKAGE;ON_LIBRARY_LINKAGE")
+    set(multi_args "ON_TARGET;ON_ARCH;ON_CRT_LINKAGE;ON_LIBRARY_LINKAGE")
+    cmake_parse_arguments(PARSE_ARGV 0 "arg" "ALWAYS" "MESSAGE" "${multi_args}")
     if(DEFINED arg_UNPARSED_ARGUMENTS)
         message(FATAL_ERROR "vcpkg_fail_port_install was passed extra arguments: ${arg_UNPARSED_ARGUMENTS}")
     endif()
+
+    if(arg_ALWAYS)
+        vcpkg_list(SET extra_args)
+        foreach(arg IN LISTS multi_args)
+            if(DEFINED "arg_${arg}")
+                vcpkg_list(APPEND extra_args "${arg}" "${arg_${arg}}")
+            endif()
+        endif()
+        if(NOT "${extra_args}" STREQUAL "")
+            message(WARNING "vcpkg_fail_port_install(ALWAYS) was passed extra arguments: ${extra_args}")

```

I feel like this doesn't explain the reason for the warning too well. How about something like

```suggestion
            message(WARNING "vcpkg_fail_port_install set to fail both unconditionally and conditionally on ${extra_args}. This is likely to be an error.")
```